packages/ra-ui-materialui/src/form/SimpleFormIterator.js

```diff
@@ -64,6 +64,8 @@ const useStyles = makeStyles(
 
 const SimpleFormIterator = props => {
     const {
+        addButton,
+        removeButton,
```

Comment:
```
        addButton = <DefaultAddButton />,
        removeButton = <DefaultRemoveButton />,
```

Reply:
Can't we do this in default props instead?

Reply:
Nope as they will probably be removed in a future react version

Reply:
Didn't knew that, thanks

Reply:
While implementing this I realized that, if we are going to keep these `<DefaultAddButton />` & `<DefaultRemoveButton />` outside the `SimpleFormIterator` then we also need to pass a few things (` source, removeField/addField, index`) through props. I think this would make things more complex.

Reply:
You shouldn't have to pass any of them. The event handlers I described before could take not only the original `onClick` handler but also the index then return a new handler. Buttons would only have to call their `onClick` prop as usual

Reply:
What about the `source`? it is used in the className property.
Also the index is also used in the className property of removeButton.

Reply:
Builds the className when you clone the element and pass it as a prop too ?

Reply:
Alright, initially the `<DefaultAddButton />` & `<DefaultRemoveButton />` will not have any className and onClick property. As it will be passed as a prop at the cloning part.
Let me know if this needs any further updates?